i18n/bundle/bundle.go

```diff
@@ -78,34 +78,79 @@ func (b *Bundle) ParseTranslationFileBytes(filename string, buf []byte) error {
 }
 
 func parseTranslations(filename string, buf []byte) ([]translation.Translation, error) {
+	// if it's standard format
+	var standardFormat []map[string]interface{}
+	err := unmarshal(filename, buf, &standardFormat)
+	if err == nil {
+		translations, err := parseStandardFormat(standardFormat)
+		if err != nil {
+			return nil, err
+		}
+		return translations, nil
+	}
+
+	// if it's flatter format
+	var flatterFormat map[string]map[string]interface{}
+	if err := unmarshal(filename, buf, &flatterFormat); err != nil {
+		return nil, err
+	}
+	translations, err := parseFlatterFormat(flatterFormat)
+	if err != nil {
+		return nil, err
+	}
+	return translations, nil
+}
+
+func unmarshal(filename string, buf []byte, out interface{}) error {
 	var unmarshalFunc func([]byte, interface{}) error
 	switch format := filepath.Ext(filename); format {
 	case ".json":
 		unmarshalFunc = json.Unmarshal
 	case ".yaml":
 		unmarshalFunc = yaml.Unmarshal
 	default:
-		return nil, fmt.Errorf("unsupported file extension %s", format)
+		return fmt.Errorf("unsupported file extension %s", format)
 	}
 
-	var translationsData []map[string]interface{}
 	if len(buf) > 0 {
-		if err := unmarshalFunc(buf, &translationsData); err != nil {
-			return nil, fmt.Errorf("failed to load %s because %s", filename, err)
-		}
+		return unmarshalFunc(buf, out)
```

Comment:
can you add back the error message? It is nice to know which file the error happened in.

```
if err := unmarshalFunc(buf, &translationsData); err != nil {
	return fmt.Errorf("failed to load %s because %s", filename, err)
}
```